merino/curated_recommendations/corpus_backends/corpus_api_backend.py

```
@@ -281,7 +287,7 @@ async def _revalidate_cache(self, surface_id: ScheduledSurfaceId) -> list[Corpus
             try:
                 data = await self.retry_fetch(surface_id, cache_key)
                 return data
-            except (HTTPError, ValueError) as e:
+            except (BackendError, HTTPError, ValueError) as e:
```

cc @katerinachinnappan : I'm not clear here how these changes interact with the recent changes you've made to error handling and caching in the endpoint, probably worth a close look from you.

---

we missed that corpus API / graphql sub-graphs mask errors, returns a 200 status code with an error message. In this case `HTTPError` would only get caught on the gateway level, the 502 we have seen. that also explains why [these errors](https://mozilla.sentry.io/issues/5623243151/?project=6622434&referrer=project-issue-stream) in Merino were not handled, because they weren't treated as `HTTPErrors`. I think we should still catch `HTTPError`, `ValueError`. Handling GraphQL errors with `BackendError` seems ok, though I would instead make a derived class `CorpusAPIBackendError` from `BackendError`:

`class CorpusAPIBackendError(BackendError)`. Merino does this already for another backend I think. & would raise & catch `CorpusAPIBackendError` to be a bit more specific.